common/src/main/java/org/opensearch/ml/common/transport/connector/MLUpdateConnectorRequest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.connector;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+@Getter
+public class MLUpdateConnectorRequest extends ActionRequest {
+    String connectorId;
+    Map<String, Object> updateContent;
```

Comment:
How about using `Connector newConnector` ?

Reply:
Define a new connector would require the user to redefine the whole connector body. We should make it easier just like how the Update API works in the OS Document. User only needs to put the updated/new/deleted field in the request like rotating the credential key, and we will update the connector in the index accordingly.

Reply:
I think we can use same way of current create connector https://github.com/opensearch-project/ml-commons/blob/2.x/plugin/src/main/java/org/opensearch/ml/action/connector/TransportCreateConnectorAction.java#L96

Reply:
Can I just do the following to update using the current create connector? or I have to reenter the whole connector template?

POST /_plugins/_ml/connectors/_update/xxxx
{
   "credential": {
        "openAI_key": "new_key"
    }
}